core/types/tx_setcode.go

```diff
@@ -179,8 +179,13 @@ func (tx *SetCodeTx) copy() TxData {
 }
 
 // accessors for innerTx.
-func (tx *SetCodeTx) txType() byte           { return SetCodeTxType }
-func (tx *SetCodeTx) chainID() *big.Int      { return tx.ChainID.ToBig() }
+func (tx *SetCodeTx) txType() byte { return SetCodeTxType }
+func (tx *SetCodeTx) chainID() *big.Int {
```

Comment:
it's impossible to have nil chainID if the tx is constructed from the json unmarshal or rlp decode, right?

it's only possible for the manual constructed one.

Reply:
Yep its only possible for hand-created transactions, I added it, because we have this comment here:
```
// ChainId returns the EIP155 chain ID of the transaction. The return value will always be
// non-nil. For legacy transactions which are not replay-protected, the return value is
// zero.
func (tx *Transaction) ChainId() *big.Int {
	return tx.inner.chainID()
}
```
I can revert, but then I would also remove this comment